sklearn/utils/estimator_checks.py

```diff
@@ -360,55 +366,242 @@ def check_transformer_pickle(name, Transformer):
+
     else:
-        alg.partial_fit(X, y)
+        ret_val = estimator.partial_fit(X)
+        assert_same_model(test_X, estimator, cloned.partial_fit(X), msg)
+
+    # Check if returns self
+    msg = "partial_fit of estimator %s did not return self" % name
+    assert_equal(ret_val, estimator, msg=msg)
 
-    assert_raises(ValueError, alg.partial_fit, X[:, :-1], y)
 
+def check_partial_fit_reset_when_fit(name, Estimator):
+    """Check if the estimators get reset when fit is applied after a
+    partial fit.
+    """
+
+    X, y = make_blobs(n_samples=50, random_state=0)
+    X -= X.min()
+
+    test_X, test_y = make_blobs(n_samples=50, random_state=1)
+    test_X -= test_X.min()
+
+    with warnings.catch_warnings(record=True):
+        estimator_1 = Estimator()
+        estimator_2 = Estimator()
+
+    set_random_state(estimator_1)
+    set_fast_parameters(estimator_1)
+
+    set_random_state(estimator_2)
+    set_fast_parameters(estimator_2)
+
+    # Initial call to fit
+    if isinstance(estimator_1, (ClassifierMixin, RegressorMixin)):
+        estimator_1.fit(X, y)
+        estimator_2.fit(X, y)
+    else:
+        estimator_1.fit(X)
+        estimator_2.fit(X)
+
+    # Run 2 iterations of full set of partial_fit with batches of size 10
+    classes = np.unique(y)
+    for i in range(2):
+        X_s, y_s = shuffle(X, y, random_state=i)
+        for j in range(0, 50, 10):
+            if isinstance(estimator_2, ClassifierMixin):
+                estimator_2.partial_fit(X_s[j:10+j], y_s[j:10+j],
+                                        classes=classes)
+            elif isinstance(estimator_2, RegressorMixin):
+                estimator_2.partial_fit(X_s[j:10+j], y_s[j:10+j])
+            else:
+                estimator_2.partial_fit(X_s[j:10+j])
+
+    # Use fit(x, y) if classifier / regressor.
+    if isinstance(estimator_1, (ClassifierMixin, RegressorMixin)):
+        # Test that change in features does not raise any error
+        assert_not_raises(Exception, estimator_2.fit, X[:, :-1], y)
+        # Finally call fit to reset the estimator
+        estimator_2.fit(X, y)
+
+    # Use fit(x) otherwise.
+    else:
+        # Test that change in features does not raise any error
+        assert_not_raises(Exception, estimator_2.fit, X[:, :-1])
+        # Finally call fit to reset the estimator
+        estimator_2.fit(X)
+
+    # Compare with the first fit model.
+    assert_same_model(test_X, estimator_1, estimator_2)
+
+
+def check_partial_fit_effect(name, Estimator):
+
+    X, y = make_blobs(n_samples=500, n_features=4, random_state=1)
+    X = StandardScaler().fit_transform(X)
+
+    # Smaller data set for 2nd partial_fit
+    dX, dy = make_blobs(n_samples=10, n_features=4, random_state=2)
+    dX = StandardScaler().fit_transform(dX)
+
+    # Test data set to compare model
+    test_X, test_y = make_blobs(n_samples=50, n_features=4, random_state=3)
+    test_X = StandardScaler().fit_transform(test_X)
```

Comment:
I thought some estimators required scaling, ( after seeing them being scaled at a few other places )... Not sure if I am right... [ Related to https://github.com/scikit-learn/scikit-learn/pull/3907#discussion_r23316229 ]... Will investigate properly and update :)